components/scream/src/interface/scream_scorpio_interface.F90

```diff
@@ -0,0 +1,933 @@
+      if (hist_var%dimlen(dim_ii).eq.0) has_t_dim = .true.
+    end do
+
+    ! Distribute responsibility for writing cores over all PIO ranks
+    ! i.e. compute the degrees of freedom that this rank will contribute to PIO
+    if (has_t_dim) then
+      call get_compdof(length-1,hist_var%dimlen(:length-1),my_dof_len,istart,istop)
+    else
+      call get_compdof(length,hist_var%dimlen(:length),my_dof_len,istart,istop)
+    end if
+    allocate( hist_var%compdof(my_dof_len) )
+    hist_var%compdof(:my_dof_len) = (/ (ii, ii=istart,istop, 1) /)
+
+    ! Register Variable with PIO
+    ierr = PIO_def_var(pio_atm_file%pioFileDesc, trim(shortname), hist_var%dtype, hist_var%dimid(:length), hist_var%pioVar)
+    call errorHandle("PIO ERROR: could not define variable "//trim(shortname),ierr)
+
+    ! Gather the pio decomposition for this variable
+    if (has_t_dim) then
+      loc_len = max(1,length-1)
+      call get_decomp(tag,hist_var%dtype,hist_var%dimlen(:loc_len),hist_var%compdof,hist_var%iodesc)
+    else
+      call get_decomp(tag,hist_var%dtype,hist_var%dimlen,hist_var%compdof,hist_var%iodesc)
+    end if
+
+    return
+  end subroutine register_variable
+!=====================================================================!
+  subroutine get_pio_file(filename,pio_atm_file,found)
+
+    character(len=*), intent(in)   :: filename
+    type(pio_atm_output), pointer  :: pio_atm_file
+    logical, intent(out), optional :: found
+
+    type(pio_file_list), pointer :: curr => NULL(), prev => NULL()
+
+    ! Starting at the top of the current list of PIO_FILES find the next empty
+    ! one in list.
+    curr => pio_file_list_top
+    do while (associated(curr%pio_file))
+      if (trim(filename)==trim(curr%pio_file%filename)) then
+        pio_atm_file => curr%pio_file
+        if (present(found)) found = .true.
+        return
+      end if
+      prev => curr
+      curr => prev%next
+    end do
+    ! If we got this far then no pio_file with that name was found
+    if (present(found)) found = .false.
+    allocate(prev%next)
+    curr => prev%next
+    allocate(curr%pio_file)
+    pio_atm_file => curr%pio_file  
+
+  end subroutine get_pio_file
+!=====================================================================!
+  subroutine eam_history_write()
+
+    type(pio_atm_output),pointer :: pio_atm_file
+    type(hist_var_t), pointer :: var 
+    integer :: ivar, ierr, jj
+    integer :: my_dummy_value, mylen
+    integer, allocatable :: fdata_int(:)
+    real(rtype) :: my_dummy_r
+    real(rtype), allocatable :: fdata_real(:)
+    real(rtype) :: time
+    character(len=100) :: varname
+    integer :: fdata_int_1d(10), fdata_int_2d(10,3)
+    real(rtype) :: fdata_real_1d(10), fdata_real_2d(10,3), fdata_real_2d_inv(3,10)
+
+
+    call get_pio_atm_file("example_pio_structured.nc",pio_atm_file)
+    pio_atm_file%numRecs = pio_atm_file%numRecs + 1
+    time = pio_atm_file%numRecs * 300.0_rtype
+
+    ! DUMMY Array output for testing
```

Comment:
Why do we need a dummy array output? In our tests, we can register a dummy array, and check the output file. I wouldn't add dummy testing stuff in the main file, if possible.

Reply:
Yes, I wanted to keep the number of files down to 1 if possible and start to peel away the not needed code as we moved to the C++ implementation.  I'm happy to move all of this to it's own Fortran file if that would make more sense before the next merge.  Or, if we don't think anyone will be using the code we can keep it as is since it is slated to be removed in phase 2.

Reply:
Keep it, if that's easier/faster for you. Just make it clear this has to go once the work is completed.